### PR TITLE
AEA-3991 extract correlation ids for step functions

### PR DESCRIPTION
**Pull request by anthony-nhs**
## Summary

- Routine Change

### Details

- extract correlation ids for step functions
- add sonar cloud coverage
- add jest and prettier vscode plugins to dev container

### CODE REVIEW

File: .github/workflows/quality_checks.yml [hunk 1]
```diff
@@ -45,3 +47,9 @@ jobs:
 
       - name: Run Unit Tests
         run: make test
+
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (originalphil) [inline .github/workflows/quality_checks.yml:54]: Does GITHUB_TOKEN also need adding alongside SONAR_TOKEN at the top of the file, and passing-in in the same manner from the calls to quality_checks?

File: packages/splunkProcessor/src/splunkProcessor.js [hunk 2]
```diff
@@ -104,25 +104,51 @@ function transformLambdaLogEvent(logEvent) {
 function transformLogEvent(logEvent, logGroup, accountNumber) {
```
Comment (originalphil) [inline packages/splunkProcessor/src/splunkProcessor.js:104]: I wouldn't say this is very readable. How about splitting it into individual functions for lambda, stepfunction and other logs?
The whitespace is a bit inconsistent, too.

File: .github/workflows/quality_checks.yml (hunk 1 above)
Comment (anthony-nhs) [reply .github/workflows/quality_checks.yml:54]: no - GITHUB_TOKEN is automatically created when a github action runs and is available to called workflows - see https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/reusing-workflows

**Review by JackSpagnoliNHS — DISMISSED** [review]

File: .github/workflows/quality_checks.yml (hunk 1 above)
Comment (JackSpagnoliNHS) [reply .github/workflows/quality_checks.yml:54]: `secrets.GITHUB_TOKEN` is available globally afaik

**Pull request merged**